lib/commands/list.ts

```diff
@@ -0,0 +1,103 @@
+import * as fs from "fs-extra";
+import * as path from "path";
+import { ProjectConfig } from "../ProjectConfig";
+import { TemplateManager } from "../TemplateManager";
+import { Util } from "../Util";
+import { PromptSession } from "./../PromptSession";
+
+let command: {
+	[name: string]: any,
+	templateManager: TemplateManager,
+	execute: (argv: any) => Promise<void>
+};
+
+// tslint:disable:object-literal-sort-keys
+command = {
+	command: "list [framework] [type]",
+	// use aliases here, instead of alias. With single alias yargs does not build correctly argv
+	aliases: ["l"],
+	desc: "list all templates",
+	builder: {
+		framework: {
+			alias: "f",
+			describe: "Framework to list templates for",
+			type: "string"
+		},
+		type: {
+			alias: "t",
+			describe: "Project type (depends on framework)",
+			type: "string"
+		}
+	},
+	templateManager: null,
+	async execute(argv) {
+		let inProject = false;
+		const viewsGroupName = "Views";
+		if (ProjectConfig.hasLocalConfig()) {
+			const config: Config = ProjectConfig.getConfig();
+			argv.framework = config.project.framework;
+			argv.type = config.project.projectType;
+			inProject = true;
+		}
+
+		const templatesByGroup = [];
+		const controlGroups: string[] = [];
+
+		const framework: Framework = this.templateManager.getFrameworkById(argv.framework);
+		if (!framework) {
+			return Util.error("Wrong framework provided", "red");
+		}
+
+		let projectLib: ProjectLibrary;
+		if (argv.type) {
+			projectLib = command.templateManager.getProjectLibrary(argv.framework, argv.type) as ProjectLibrary;
+			if (!projectLib) {
+				return Util.error(`Project type '${argv.type}' not found in framework '${argv.framework}'`, "red");
+			}
+		} else {
+			projectLib = command.templateManager.getProjectLibrary(argv.framework) as ProjectLibrary;
+		}
+
+		let maxIdLength = 0;
+		projectLib.templates
+			.filter(t => t.controlGroup !== "Custom Templates")
```

Comment:
That "Custom Templates" seems like more of a error in the Angular Editors form template (ig-ts).